usecase/pokemon_service.go

```diff
@@ -0,0 +1,36 @@
+package usecase
+
+import (
+	"github.com/nibble-4bits/ondemand-go-bootcamp/entity"
+)
+
+type pokemonService struct {
+	repo PokemonRepository
+}
+
+type PokemonService interface {
+	GetByID(id int) (entity.Pokemon, error)
+	GetAll() ([]entity.Pokemon, error)
+}
+
+func NewPokemonService(r PokemonRepository) pokemonService {
```

Comment:
Be careful, you are returning an unexported field. This might be annoying to use.

Reply:
What would be a better practice?